src/runtime_src/core/pcie/driver/linux/xocl/lib/libxdma.c

```diff
@@ -1437,11 +1437,18 @@ static int xdma_process_requests(struct xdma_engine *engine,
 		timeout = wait_event_interruptible_timeout(req->arbtr_wait,
 				req->sw_desc_cnt == req->desc_completed,
 				msecs_to_jiffies(10000)); /* 10sec timeout */
+
+		/* Check for timeouts or errors */
 		if (timeout == 0) {
 			pr_err("Request completion timeout\n");
 			engine_reg_dump(engine);
 			rv = -EIO;
+		} else if (timeout < 0) {
+			pr_err("Request aborted with error = %d\n", timeout);
+			engine_reg_dump(engine);
+			rv = -EIO;
```

Comment:
I still have concern about this. What if the hardware completes the request after user canceled it (timeout < 0)? Will that cause any problem?
Maybe we should change wait_event_interruptible_timeout to wait_event_timeout which does not allow user cancel the request.

Reply:
if hardware completes after user cancels it  the request will still be completed with ERROR. 

 I do not see anything bad happening  in terms of corruption or panic. If it is acceptable to wait for 10 seconds after user cancels then we can make it non-interruptible.

Reply:
I think in this case the current request returns error. Then the hardware completes. Do you mean this hardware completion will be ignored gracefully and will not impact next request?

Reply:
I see. yes it can impact next request because completion for current request will "look" like completion for next request which is not true.  Although in synchronous transfers it would fine but in asynchronous transfers it can impact next request. 

You suggest I should make it  non-interruptible ?

Reply:
I suggest to just make it non-interruptible.

Reply:
The simplest is to not allow cancel as Lizhi suggested.
Otherwise, we need to keep the # descriptor used for the cancelled request (and also  still maintain the same request order) in case the request eventually get completed.